nixos/modules/services/monitoring/glpi-agent.nix

```diff
@@ -0,0 +1,96 @@
+# GLPI Agent daemon.
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.glpiAgent;
+
+  configFile = pkgs.writeText "agent.cfg" ''
+    # Server configuration
+    server = ${lib.concatStringsSep ", " cfg.servers}
+
+    # Configuration
+    ${cfg.extraConfig}
+  '';
+
+in
+{
+  options = {
+    services.glpiAgent = {
+      enable = lib.mkEnableOption "GLPI Agent";
+
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.callPackage ./package.nix { };
+        description = "The GLPI agent package to use.";
+      };
+
+      servers = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        description = ''
+          The URLs of the GLPI servers to connect to.
+        '';
+        example = [ "http://glpi.example.com/inventory" ];
+      };
+
+      extraConfig = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Configuration options that will be written to the configuration file.
+        '';
+        example = ''
+          delaytime = 3600
+          timeout = 180
+          tag = mytag
+          logger = stderr
+          # etc...
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    # User and group
+    users.users.glpi-agent = {
+      description = "GLPI Agent user";
+      isSystemUser = true;
+      group = "glpi-agent";
+      home = "/var/lib/glpi-agent";
+      createHome = true;
+    };
+
+    users.groups.glpi-agent = { };
+
+    # Create directories and manage configuration file
+    systemd.tmpfiles.rules = [
```

Comment:
Please use the new syntax with settings

Reply:
Hi,

What is the new syntax with settings ?

Best Regards

Reply:
That is correct : 

```
systemd.tmpfiles = {
      settings = {
        "glpi-agent-var" = {
          d = {
            "/var/lib/glpi-agent" = {
              mode = "0755";
              user = "glpi-agent";
              group = "glpi-agent";
            };
            "/var/lib/glpi-agent/var" = {
              mode = "0755";
              user = "glpi-agent";
              group = "glpi-agent";
            };
            "/etc/glpi-agent" = {
              mode = "0755";
              user = "glpi-agent";
              group = "glpi-agent";
            };
          };
          C = {
            "/etc/glpi-agent/agent.cfg" = {
              mode = "0644";
              user = "glpi-agent";
              group = "glpi-agent";
              argument = "${configFile}";
            };
          };
        };
      };
    };
```